Remove null-checks for WindowManagerState

msisov opened the pull request:
Two fixups that remove added null-checks for WindowManagerState.

msisov committed:
fixup! c++ / mojo changes for 'external window mode' (#9)
  -- remove WindowManagerState nullchecks as per
     https://github.com/Igalia/chromium/issues/10

msisov committed:
fixup! fixup! c++ / mojo changes for 'external window mode' (#5) (#1)
   -- Remove null-checks to WindowManagerState as per
   https://github.com/Igalia/chromium/issues/10

fred-wang commented on services/ui/ws/display.cc:
```diff
@@ -58,19 +58,17 @@ Display::~Display() {
-          window_server_->DestroyTree(tree);
-      }
+    WindowTree* tree = display_root->window_manager_state()->window_tree();
+    ServerWindow* root = display_root->root();
```
This is going to be a pain to merge. Maybe just remove the if() { ... } here and fix the indentation in a follow-up commit.

msisov replied:
Why? There is no merge conflict

msisov replied:
Divided to two different commits - remove if and fix indentation

fred-wang commented on services/ui/ws/window_tree.cc:
```diff
@@ -302,7 +302,7 @@ void WindowTree::OnWmMoveDragImageAck() {
   ServerWindow* window = GetWindowByClientId(client_window_id);
   WindowManagerDisplayRoot* display_root = GetWindowManagerDisplayRoot(window);
-  if (display_root && !display_root->window_manager_state())
+  if (display_root)
```
For this and the one in WindowTree::ReleaseCapture there was no early return at all. I guess we can remove this display_root check unless to reduce the changeset. If that check turns out to be necessary, it should probably be upstream.

fred-wang reviewed (commented):
Can you please name both commits "fixup! c++ / mojo changes for 'external window mode'" ?

msisov commented on services/ui/ws/window_tree.cc:
```diff
@@ -302,8 +302,6 @@ void WindowTree::OnWmMoveDragImageAck() {
 bool WindowTree::SetCapture(const ClientWindowId& client_window_id) {
   ServerWindow* window = GetWindowByClientId(client_window_id);
   WindowManagerDisplayRoot* display_root = GetWindowManagerDisplayRoot(window);
-  if (display_root && !display_root->window_manager_state())
```
this is fixed!

msisov committed:
fixup! fixup! (TEMP) basic event/mouse support (#1)
  -- Fix identation after fixup! c++ / mojo changes for 'external window
     mode' (#9)

fred-wang reviewed (approved):
lgtm. Please merge the PR without squashing.

tonikitoo reviewed (commented):
lgtm

tonikitoo reviewed (commented):
lgtm

tonikitoo reviewed (commented):
lgtm

tonikitoo commented:
feel free to merge them in the way you thing it works best for squashing.
